compiler/optimizer.py

```diff
@@ -44,6 +44,20 @@ def visitIfCodelet( self, if_codelet ):
         else:
             return self.visitCodelet( if_codelet )
 
+    def visitSyscallCodelet( self, syscall_codelet ):
+        name = syscall_codelet.name()
+        args = syscall_codelet.arguments()
+        if isinstance( args, codetree.SyscallCodelet ):
+            args_name = args.name()
+            if args_name == "..<" or args_name == "...":
+                transformed_args = args.arguments().visit( self )
```

Comment:
Is this "depth-first visiting" specific to ranges, or even Simplify?

Reply:
No, it's a general pattern. I assume you're suggesting that it would be a good idea to abstract the pattern further?